### ISSUE TITLE
Embedded app hang: FlutterEnginePluginRegistry: Attempted to detach plugins from an Activity when no Activity was attached.

### ISSUE THREAD

**Issue by bDev10** [flutter/flutter#47342]
I think this is very critical bug in the new stable release of flutter SDK. Previous stable release when used facade embedding classes hasn't got this issue. Now facade classes are absent, so only embedding classes can be used and this bug makes publishing apps with embedded Flutter code impossible. I think this issue should be repaired ASAP as another Hotfix to current stable Flutter SDK release.

## Steps to Reproduce

1. Create new Android project in Android Studio (for example using "Basic Activity" template). I used Android Studio 3.6 RC1 and version 3.5 too.
2. Integrate Flutter module into Android project [as described here, by manual integration Option B - Depend on the module’s source code](https://flutter.dev/docs/development/add-to-app/android/project-setup).
3. Add/embed Flutter to Android using standalone Activity [as described here](https://github.com/flutter/flutter/wiki/Experimental:-Add-Flutter-Activity) [with FlutterEngineCache method (pre-warmed)](https://github.com/flutter/flutter/wiki/Experimental:-Add-Flutter-Activity#use-flutterenginecache).
```
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        // Instantiate your FlutterEngine.
        // Instantiate your FlutterEngine.
        val flutterEngine = FlutterEngine(applicationContext)

        // Pre-warm your FlutterEngine by starting Dart execution.
        // Pre-warm your FlutterEngine by starting Dart execution.
        flutterEngine
            .dartExecutor
            .executeDartEntrypoint(
                DartEntrypoint.createDefault()
            )

        FlutterEngineCache
            .getInstance()
            .put("my_engine_id", flutterEngine);
    }
}
```
4. Start Flutter Activity when tapped on Main Activity's feature action button or any other.
```
Intent intent = FlutterActivity
  .withCachedEngine("my_engine_id")
  .build(context);
startActivity(intent);
```
5. Build and launch app.
6. Tap on feature action button to show flutter activity (counter).
7. Tap on devices "back" button to go back to main activity.
8. Go to point 6 above until the app hang in Flutter activity and back button stops working. Be patient, sometimes hangs after only few reopenings, but usually I need to go to point 6 about 50 - 100 times.

**Target Platform:**
Android. Flutter embedded in existing Android app.

**Target OS versions and devices:**
- Xiaomi Redmi 6A, Android 8.1
- Motorola Moto x4 XT1900-7, Android 9
- Google Nexus 9, Android 7.1.1
- Google Nexus 5X, Android 8.1.0
I think the bug appears more often on specific devices (in my case the listed above, from many others I have), but probably it can appear on any device and OS version.

## Logs

```
2019-12-18 14:48:34.536 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:34.536 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:34.536 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:34.561 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:34.571 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:34.580 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:34.656 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x7227594940
2019-12-18 14:48:35.251 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:35.255 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:35.483 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:35.483 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:35.483 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:35.499 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:35.522 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:35.530 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:35.643 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x7227594920
2019-12-18 14:48:36.136 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:36.141 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:36.245 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:36.245 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:36.246 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:36.267 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:36.281 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:36.292 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:36.392 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x7227594dc0
2019-12-18 14:48:36.721 5592-5597/com.example.embedder I/art: Do partial code cache collection, code=61KB, data=58KB
2019-12-18 14:48:36.721 5592-5597/com.example.embedder I/art: After code cache collection, code=61KB, data=58KB
2019-12-18 14:48:36.721 5592-5597/com.example.embedder I/art: Increasing code cache capacity to 256KB
2019-12-18 14:48:37.030 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:37.033 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:37.179 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:37.179 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:37.179 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:37.194 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:37.212 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:37.218 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:37.315 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x72296fd900
2019-12-18 14:48:37.907 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:37.912 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:38.015 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:38.016 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:38.016 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:38.032 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:38.166 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x72296fdc00
2019-12-18 14:48:38.701 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:38.704 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:38.778 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:38.778 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:38.778 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:38.796 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:38.817 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:38.825 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:38.918 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x72297ccfc0
2019-12-18 14:48:39.446 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:39.450 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:39.461 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:39.461 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:39.461 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:39.482 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:39.504 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:39.511 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:39.607 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x722c034780
2019-12-18 14:48:40.060 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:40.060 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:40.060 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:40.078 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:40.087 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:40.096 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:40.226 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x722c05e460
2019-12-18 14:48:40.714 5592-5592/com.example.embedder D/FlutterActivity: Using the launch theme as normal theme.
2019-12-18 14:48:40.714 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Setting up FlutterEngine.
2019-12-18 14:48:40.715 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Attaching FlutterEngine to the Activity that owns this Fragment.
2019-12-18 14:48:40.738 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:40.756 5592-5592/com.example.embedder W/FlutterView: Tried to send viewport metrics from Android to Flutter but this FlutterView was not attached to a FlutterEngine.
2019-12-18 14:48:40.764 5592-5592/com.example.embedder D/FlutterView: Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:40.894 5592-5619/com.example.embedder D/OpenGLRenderer: endAllActiveAnimators on 0x722cc43400 (RippleDrawable) with handle 0x722c11b140
2019-12-18 14:48:41.176 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:41.182 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:41.190 5592-5592/com.example.embedder D/FlutterView: Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@a6102d8
2019-12-18 14:48:41.195 5592-5592/com.example.embedder D/FlutterActivityAndFragmentDelegate: Detaching FlutterEngine from the Activity that owns this Fragment.
2019-12-18 14:48:41.195 5592-5592/com.example.embedder E/FlutterEnginePluginRegistry: Attempted to detach plugins from an Activity when no Activity was attached.
```

**flutter analyze**
```
Analyzing SimpleEmbedding...
No issues found! (ran in 4.3s)
```

**flutter doctor -v**
```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.535], locale pl-PL)
    • Flutter version 1.12.13+hotfix.5 at c:\Users\BB\AppData\Local\Android\flutter
    • Framework revision 27321ebbad (8 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\BB\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\Users\BB\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio Beta\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio Beta
    • Flutter plugin version 42.1.3
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] Connected device (1 available)
    • Nexus 9 • HT4C9JT00640 • android-arm64 • Android 7.1.1 (API 25)

• No issues found!
```

**Comment by lawonga** [comment]
Running into this issue too when I rapidly launch/close flutter screens

**Comment by bDev10** [comment]
Could someone from Flutter team be assigned to take closer look at this issue? Actually it is a blocker to release app with embedded Flutter activity :(.

**Comment by AllenWen** [comment]
I got the same problem,cannot reopen flutteractivity after using providedcachedengine

**Comment by dieasquo** [comment]
i got same issue, i use Pixel 2 Api 24. after pressing the device rotation igot this error.. hope it can be fixed as soon as possible

**Comment by omprakash980** [comment]
anyone can find the solution to that , to me its like D/FlutterView( 9517): Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@6148da8, the process of installation is slow

**Comment by hirakjyotiborah** [comment]
Encoutering the exact same problem

**Comment by dieasquo** [comment]
maybe this can help, just to IGNORING the error not solved completely.
so you will not face the black screen when debug your application.

1. download new SDK Platform in Android Studio > SDK Manager > Android SDK > SDK Platform.

2. Choose Android API 23 (Marshmallow) or API 24 (Nougat). 
 *you can try another API Version but, i recommend Api 23 / 24*
 and Apply to download

3. Open AVD Manager > Create Virtual  Device > Choose Nougat (if you dont have it, please download first) >  And Click Finish

4. debug app with your new Virtual Device

Maybe its just create new virtual device may solve your problem. so dont touch the rotation button. if you want to check the Function, i advice using Physical devices. 

hope it can little helping you

**Comment by bDev10** [comment]
@dieasquo This problem exists on physical devices with any API version and the issue is that it should not occur on published production versions of apps. Avoiding a crash on developer device/emulator for example by not rotating a device to be able to test the app is not a problem.

**Comment by AllenWen** [comment]
seems THE OFFICIAL is not trying to solve this problem, but there is other cached engined resolution like [flutter-boost](https://github.com/alibaba/flutter_boost)

**Comment by bDev10** [comment]
@AllenWen Did using flutter-boost workarounded the problem for you?

**Comment by ahammer** [comment]
We use Flutter Embedded and saw this in our logs. 

We have our own FlutterActivity, with SingleTop to enforce that there is only one instance of it. However we'd find that if you went back and forth fast, the Activity would get multiple instances and they'd fight the life-cycle. Eventually you'd get left in a inconsistent state.

To fix it, in our FlutterActivity we track a weakreference to the activity from start to stop. We check if the weakreference is null before a) Launching the Intent and b) Finishing onCreate (calling finish instead of another activity is detected). We also throttled how fast a user can click to get to a flutter screen, so that once they click and load a screen they need to wait 500ms before launching the activity again. 

By preventing monkey-touches from launching the FlutterActivity twice quickly, and by checking that it's the only started instance, it seems we are able to avoid this issue.

**Comment by neeboo** [comment]
I have same problem using custom FlutterActivity, especially when the activity is popped.

But somehow I have some **possible** solution like this:

```kotlin
class MyCustomActivity: FlutterActivity(){
  
   
  ...
    @SuppressLint("VisibleForTests")
    override fun onResume() {
        super.onResume()
        super.delegate.onAttach(this)
    }

    @SuppressLint("VisibleForTests")
    override fun onPostResume() {
        super.onPostResume()
        super.delegate.onAttach(this)
    }

   // Better add this as well
   override fun shouldAttachEngineToActivity(): Boolean {
        return true
    }
   ...

}
```

It seems we have to reattach the activity,

**Comment by TahaTesser** [comment]
Hi @bDev10 and everyone
I can't reproduce this issue no matter how many times I try on the latest `master` on Realme 6 running Android 10

<details> 
<summary>complete code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}
```
```kt
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import io.flutter.embedding.android.FlutterActivity


class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val buttonCachedEngine = findViewById<Button>(R.id.launch_flutter_cached_engine)
        buttonCachedEngine.setOnClickListener {
            val intent = FlutterActivity
                .withCachedEngine("my_engine_id")
                .build(this)
            startActivity(intent)
        }


    }
``` 

```kt
import android.app.Application
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache
import io.flutter.embedding.engine.dart.DartExecutor

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        // Instantiate your FlutterEngine.
        // Instantiate your FlutterEngine.
        val flutterEngine = FlutterEngine(applicationContext)

        // Pre-warm your FlutterEngine by starting Dart execution.
        // Pre-warm your FlutterEngine by starting Dart execution.
        flutterEngine
            .dartExecutor
            .executeDartEntrypoint(
                    DartExecutor.DartEntrypoint.createDefault()
            )

        FlutterEngineCache
            .getInstance()
            .put("my_engine_id", flutterEngine);
    }
}
```
```xml
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical"
    android:layout_gravity="center"
    android:gravity="center"
    tools:context=".MainActivity">


    <Button
        android:id="@+id/launch_flutter_cached_engine"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginBottom="20dp"
        android:layout_gravity="center"
        android:gravity="center"
        android:text="@string/launch_flutter_cached_engine" />

</LinearLayout>
```

</details>


<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.25.0-5.0.pre.108, on Microsoft Windows [Version 10.0.19042.630], locale en-US)
    • Flutter version 1.25.0-5.0.pre.108 at C:\Code\flutter_master
    • Framework revision 995aa138cd (9 hours ago), 2020-12-08 09:18:02 +0530
    • Engine revision b0d31ae166
    • Dart version 2.12.0 (build 2.12.0-125.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30717.126
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.2.4
    • Flutter plugin version 52.0.5
    • Dart plugin version 203.5981.152

[✓] VS Code (version 1.51.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19042.630]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 87.0.664.52

• No issues found!
```

</details>

Can anyone please provide an updated reproducible sample and steps that reproduce the issue on the latest version of Flutter?
Thanks

**Comment by bDev10** [comment]
Hi @TahaTesser
Indeed on the actual master (version 1.25.0-9.0.pre.24) I cannot reproduce it too, however on the actual stable (version 1.22.5) the issue still exists. So the hope  is that next stable version release will have this bug fixed, thankt you :)

**Comment by xster** [comment]
Since we're talking about rapid closing and opening etc, I have some suspicion it's https://github.com/flutter/flutter/issues/66192. The timing seems to match too where that patch first landed in 1.23. I'll tentatively close this issue. If you're still encountering this issue on versions 1.23 and above, please reopen.
